mergesort.c

```diff
@@ -102,6 +102,15 @@ bool validate(List* source)
     return true;
 }
 
+void freeList(List* source)
+{
```

Comment:
Once rewritten in `for` statement, it becomes iterator-form:
```C
for (; source; source = source->next) free(source);
```

Reply:
Pretty simplicity!

Reply:
After a long time ago, I suddenly find this statement doesn't work. 
Before the next iteration starts, ```source``` will be freed, which induces  ```for```statement terminated.